strawberryfields/apps/qchem/vibronic.py

```diff
@@ -138,6 +141,42 @@ def energies(samples: list, w: np.ndarray, wp: np.ndarray) -> Union[list, float]
     return [np.dot(s[: len(s) // 2], wp) - np.dot(s[len(s) // 2 :], w) for s in samples]
 
 
+def VibronicTransition(U1: np.ndarray, r: np.ndarray, U2: np.ndarray, alpha: np.ndarray):
+    r"""An ``sf`` operation for applying the Doktorov operator
+    :math:`\hat{U}_{\text{Dok}} = \hat{D}({\alpha}) \hat{R}(U_2) \hat{S}({r}) \hat{R}(U_1)` on a
+    given state.
```

Comment:
Maybe worth adding a sentence explaining the connection between Doktorov and VibronicTransition, e.g. "The Doktorov operator describes the transformation of the vibrational modes of a molecule when it undergoes a transition from an initial to a final electronic state"